### ISSUE TITLE
Validation is synchronous and blocks single-user server

### ISSUE THREAD

**Issue by rkdarst** [jupyter/nbgrader#1232]
When validating via the notebook interface ("validate" button), it makes a synchronous HTTP request.  This blocks the entire single-user notebook server, so that even trying to go to another notebook doesn't work.  It stays blocked even after you refresh the page - and blocks all other nbserver interaction, too.

In our courses, some instructors do extensive computation in validation, so that I've had to increase the validation timeout to 240 seconds (and incidentally, all the HTTP ingress proxy timeouts, too).  So this isn't a purely theoretical issue... though instructors probably should be encouraged to make validation a faster anyway.

I'm not sure it's worth making this an urgent issue, but submitting this to document it.

### Operating system
Ubuntu 18.04
### `nbgrader --version`
recent master (newer than 0.6.0)
### `jupyter notebook --version` 
6.0.0

**Comment by jhamrick** [comment]
Yeah, that seems undesirable. I think it's worth trying to fix this so it doesn't block.
